File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/utility/OneToManyMessageFormatter.java
package org.nirvanasms.api.core.utility;

import org.nirvanasms.api.core.base.FormatterBase;
import org.nirvanasms.api.core.base.MessageBase;
import org.nirvanasms.api.core.entity.OneToManyMessage;
import org.nirvanasms.api.core.entity.PhoneNumber;

public class OneToManyMessageFormatter extends FormatterBase<OneToManyMessage,MessageBase> {
	@Override
	public OneToManyMessage format(MessageBase message){
		PhoneNumberFormatter phoneNumberFormatter = new PhoneNumberFormatter();
        OneToManyMessage _message = (OneToManyMessage)message ;
        for(PhoneNumber number : _message.getGsmNumbers()){
        	phoneNumberFormatter.format(number);
        }
		return _message;
	}
}

File: NirvanaSMS.Api.CSharp/LibraryProject/Core/Utility/OneToManyMessageFormatter.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;

namespace NirvanaSMS.Api.Core.Utility
{
    public class OneToManyMessageFormatter : FormatterBase<OneToManyMessage,MessageBase>
    {
        public override OneToManyMessage Format(MessageBase message)
        {
            PhoneNumberFormatter phoneNumberFormatter = new PhoneNumberFormatter();
            OneToManyMessage _message = message as OneToManyMessage;
            _message.GsmNumbers.ForEach(x => x = phoneNumberFormatter.Format(x));
            //message.Message.Trim();
            return _message;
        }
    }
}
